src/helm/proxy/clients/palmyra_client.py

```diff
@@ -0,0 +1,184 @@
+        return result
+
+    def make_request(self, request: Request) -> RequestResult:
+        """Make a request"""
+        raw_request = {
+            "prompt": request.prompt,
+            "minTokens": "1",  # Setting to 1 for now
+            "maxTokens": request.max_tokens,
+            "temperature": request.temperature,
+            "topP": request.top_p,
+            "topK": request.top_k_per_token,
+            "stop": request.stop_sequences,
+            "presencePenalty": request.presence_penalty,
+            "frequencyPenalty": request.frequency_penalty,
+            # bestOf: figure out what this does
+        }
+
+        completions: List[Sequence] = []
+        model_name: str = request.model_engine
+
+        # `num_completions` is not supported, so instead make `num_completions` separate requests.
+        for completion_index in range(request.num_completions):
+            try:
+
+                def do_it():
+                    result = self._send_request(model_name, raw_request)
+                    assert "choices" in result, f"Invalid response: {result}"
+                    return result
+
+                # We need to include the engine's name to differentiate among requests made for different model
+                # engines since the engine name is not included in the request itself.
+                # In addition, we want to make `request.num_completions` fresh
+                # requests, cache key should contain the completion_index.
+                # Echoing the original prompt is not officially supported by Anthropic. We instead prepend the
+                # completion with the prompt when `echo_prompt` is true, so keep track of it in the cache key.
+                cache_key = Client.make_cache_key(
+                    {
+                        "engine": request.model_engine,
+                        "echo_prompt": request.echo_prompt,
+                        "completion_index": completion_index,
+                        **raw_request,
+                    },
+                    request,
+                )
+
+                response, cached = self.cache.get(cache_key, wrap_request_time(do_it))
+            except (requests.exceptions.RequestException, AssertionError) as e:
+                error: str = f"PalmyraClient error: {e}"
+                return RequestResult(success=False, cached=False, error=error, completions=[], embedding=[])
+
+            # Post process the completion.
+            response_text: str = response["choices"][0]["text"]
+
+            # The Anthropic API doesn't support echo. If `echo_prompt` is true, combine the prompt and completion.
+            text: str = request.prompt + response_text if request.echo_prompt else response_text
+            # The Anthropic API doesn't return us tokens or logprobs, so we tokenize ourselves.
+            tokenization_result: TokenizationRequestResult = self.tokenize(
```

Comment:
I would prefer if we passed in a `HuggingFaceClient` like [this](https://github.com/stanford-crfm/helm/blob/a0a820c4b764c3e98394b0e635e11f6620affb32/src/helm/proxy/clients/auto_client.py#L88), and then call tokenize using that client.

Reply:
Also, change Anthropic -> Writer in the comments.

Reply:
Done